app/ui-react/syndesis/src/modules/data/shared/ViewSqlForm.tsx

```diff
@@ -0,0 +1,192 @@
+}
+
+export interface IViewSqlFormState {
+  queryResults: QueryResults;
+  querySql: string;
+  viewName: string;
+}
+
+export class ViewSqlForm extends React.Component<
+  IViewSqlFormProps,
+  IViewSqlFormState
+> {
+  public static queryResultsEmpty: QueryResults = {
+    columns: [],
+    rows: [],
+  };
+
+  public constructor(props: IViewSqlFormProps) {
+    super(props);
+    this.state = {
+      queryResults: ViewSqlForm.queryResultsEmpty,
+      querySql:
+        'SELECT * FROM ' +
+        (this.props.viewNames.length > 0
+          ? this.props.viewNames[0]
+          : '<view name>'),
+      viewName: this.props.viewNames.length > 0 ? this.props.viewNames[0] : '',
+    };
+
+    this.updateQueryResults = this.updateQueryResults.bind(this);
+    this.viewSelected = this.viewSelected.bind(this);
+  }
+
+  public updateQueryResults(results: QueryResults) {
+    {
+      results && results.columns.length > 0
+        ? this.setState({ queryResults: results })
+        : this.setState({ queryResults: ViewSqlForm.queryResultsEmpty });
+    }
+
+    this.props.onQueryResultsChanged(this.state.queryResults);
+  }
+
+  public viewSelected() {
+    // TODO: change the defaultValue in the SqlStatement text area
+    // to reflect the newly selected view name (if changed)
+    // Maybe even show a text field showing the selected component depending on the PF4 widget type
+  }
+
+  public render() {
+    const formDefinition = {
+      rowLimit: {
+        componentProperty: true,
+        defaultValue: '10',
+        deprecated: false,
+        description: i18n.t(
+          'data:virtualization.viewSqlFormRowLimitDescription'
+        ),
+        displayName: i18n.t('data:virtualization.viewSqlFormRowLimit'),
+        javaType: 'java.lang.Integer',
+        kind: 'property',
+        order: 3,
+        required: true,
+        secret: false,
+        type: 'number',
+      },
+      rowOffset: {
+        componentProperty: true,
+        defaultValue: '1',
+        deprecated: false,
+        description: i18n.t(
+          'data:virtualization.viewSqlFormRowOffsetDescription'
+        ),
+        displayName: i18n.t('data:virtualization.viewSqlFormRowOffset'),
+        javaType: 'java.lang.Integer',
+        kind: 'property',
+        order: 4,
+        required: false,
+        secret: false,
+        type: 'number',
+      },
+      sqlStatement: {
+        componentProperty: true,
+        defaultValue: this.state.querySql,
+        deprecated: false,
+        description: i18n.t(
+          'data:virtualization.viewSqlFormSqlStatementDescription'
+        ),
+        displayName: i18n.t('data:virtualization.viewSqlFormSqlStatement'),
+        height: 300,
+        javaType: 'java.lang.String',
+        kind: 'property',
+        order: 2,
+        required: true,
+        secret: false,
+        type: 'textarea',
+      },
+    } as IFormDefinition;
+
+    return (
+      <WithVirtualizationHelpers username="developer">
```

Comment:
should probably put a TODO here to get the real username.  there are other places that we'll need to fix eventually too.

Reply:
see below